docker/prepare_proxy.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+# Linkerd initialization script responsible for setting up port forwarding.
+# Based on: https://github.com/istio/pilot/blob/pilot-0-2-0-working/docker/prepare_proxy.sh
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+usage() {
+  echo "${0} -p PORT"
+  echo ''
+  echo '  -p: Specify the linkerd daemonset port to which redirect all TCP traffic'
+  echo ''
+}
+
+while getopts ":p:m:" opt; do
+  case ${opt} in
+    p)
+      LINKERD_PORT=${OPTARG}
+      ;;
+    m)
+      RUN_IN_MINIKUBE=${OPTARG}
+      ;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      usage
+      exit 1
+      ;;
+  esac
+done
+
+if [[ -z "${LINKERD_PORT-}" ]]; then
+  echo "Please set linkerd port -p"
+  usage
+  exit 1
+fi
+
+# Don't forward local traffic
+iptables -t nat -A OUTPUT -d 127.0.0.1/32 -j RETURN                                         -m comment --comment "istio/bypass-explicit-loopback"
+
+if [ "$RUN_IN_MINIKUBE" = true ]; then
+  # Forward traffic to the linkerd service VIP on the single minikube node
+  iptables -t nat -A OUTPUT -p tcp -j DNAT --to ${L5D_SERVICE_HOST}:${LINKERD_PORT}         -m comment --comment "istio/dnat-to-minikube-l5d"
```

Comment:
Kubernetes creates these env variables (https://kubernetes.io/docs/concepts/containers/container-environment-variables/)
This does assume we've named the linkerd service "l5d" - so I should probably explicitly ask for the host